pkg/controllers/reconciler.go

```diff
@@ -175,6 +177,18 @@ func UpgradeImages(image *string, initimage *string) (upgrade bool) {
 
 		if parts := strings.SplitN(*s, ":", 2); len(parts) == 2 && len(parts[0]) > 0 {
 			name, version := parts[0], parts[1]
+
+			envVarValue := os.Getenv(strings.ReplaceAll(strings.ToUpper(filepath.Base(name)), "-", "_") + "_SHA")
+
+			if envVarValue != "" && *s != envVarValue {
+				log.Log.Info("env var for the image: " + name + " is already set; user input of the image is ignored")
```

Comment:
pass in `ctx` and use the logger instance from Context.

Reply:
fixed!